### ISSUE TITLE
[@storybook/addon-a11y]: Critical dependency issue

### ISSUE THREAD

**Issue by fcisio** [storybookjs/storybook#14321]
**Describe the bug**
Enabling `@storybook/addon-a11y` logs this dependency error.

> ⚠️ Critical dependency: require function is used in a way in which dependencies cannot be statically extracted

- The issue seems to be present at least from `6.1.19` to `6.1.21`.
- Disabling the addon fixes the issue.
- While the error is logged, it doesn't seem to affect the app

**Expected behavior**
The console should run without errors

**Screenshots**
![ScreenShot 2021-03-23 at 5 04 51 PM](https://user-images.githubusercontent.com/46095852/112218936-bd5cea00-8bfa-11eb-83e3-a17201fa1467.png)

**System**
System:
    OS: macOS 11.1
    CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
  Binaries:
    Node: 14.15.4 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.14.10 - /usr/local/bin/npm
  Browsers:
    Chrome: 89.0.4389.90
    Firefox: 86.0.1
    Safari: 14.0.2

**Comment by PLSFIX** [comment]
Same

**Comment by lurch83** [comment]
Also getting this issue.

**Comment by shilman** [comment]
If you want this fixed, please upvote by adding a 👍 to the issue description. We use this to help prioritize!

**Comment by mts** [comment]
- Reproducible here through the following steps using Node v14.16.0, NPM v6.14.4 and Storybook v6.1.21: 
  - https://github.com/mts/freestyler
    - `git clone git@github.com:mts/freestyler.git`
    - `npm run install-packages`
    - `npm run bootstrap`
    - `npm run start-storybook`
- ![image](https://user-images.githubusercontent.com/28991521/112630275-a0f8c180-8e35-11eb-9df3-41fd2d3f8e33.png)

**Comment by shilman** [comment]
Try upgrading to the latest prerelease:

```
npx sb@next upgrade --prerelease
```

Does that fix it?

**Comment by mts** [comment]
@shilman Yes, upgrading to the latest prerelease fixes it.
![image](https://user-images.githubusercontent.com/28991521/112645619-effb2280-8e46-11eb-937f-eeb305d68c75.png)

**Comment by panjiesw** [comment]
The warning still comes up in `v6.2.3` for `/node_modules/axe-core/axe.js`, not in CLI console when running `start-storybook`, but in browser console as well as when running `build-storybook`.

Browser console:

![2021-04-06-164319_4480x1440_scrot](https://user-images.githubusercontent.com/1451848/113692230-8822c280-96f7-11eb-8560-e5ba3972f3a4.png)

`build-storybook`:

![2021-04-06-164456_4480x1440_scrot](https://user-images.githubusercontent.com/1451848/113692257-91139400-96f7-11eb-8af2-6146dd2105f5.png)

**Comment by shilman** [comment]
@panjiesw Do you (or anybody else) have a repro repo they can share that uses 6.2.x?

**Comment by mts** [comment]
@shilman I have a repro [repo](https://github.com/mts/freestyler) using the latest Storybook v6.2.3, but the critical dependency warning has already disappeared since v6.2.0 and Storybook build looks good.

**Comment by shilman** [comment]
Ok thanks! I'm going to call this closed until somebody produces a 6.2.3+ repro.

**Comment by panjiesw** [comment]
@shilman turns out it was because of issue in `axe-core@4.1.3` https://github.com/dequelabs/axe-core/pull/2843

Tried to reproduce using fresh install, the issue disappeared because there is a new release of `axe-core@4.1.4` that resolves this.
